sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -995,9 +1099,37 @@ def create(cls, environment, state, provision_info):
         provision_info)
 
 
+class MultiWorkerWorkerHandler(object):
+  """ This class handles multi worker handlers and round robin tasks to
+  workers."""
+
+  def __init__(self):
+    self._worker_handlers = {}
+
+  def _add_worker_handler(self, worker_handler):
+    self._worker_handlers[worker_handler._worker_id] = worker_handler
+
+  def _get_worker_handler(self, worker_id=None):
```

Comment:
I'd split this into two methods, rather than having a variant return type based on whether worker_id is set.